arch/toolchains/ecmwf-hpc2020-intel-sycl.cmake

```diff
@@ -0,0 +1,163 @@
+set( EC_HAVE_EXECINFO_BACKTRACE 1 )
+set( EC_HAVE_GMTIME_R 1 )
+set( EC_HAVE_GETPWUID_R 1 )
+set( EC_HAVE_GETPWNAM_R 1 )
+set( EC_HAVE_READDIR_R 1 )
+set( EC_HAVE_DIRENT_D_TYPE 1 )
+set( EC_HAVE_GETHOSTBYNAME_R 1 )
+set( EC_HAVE_ATTRIBUTE_CONSTRUCTOR 1 )
+set( EC_ATTRIBUTE_CONSTRUCTOR_INITS_ARGV 0 )
+set( EC_HAVE_PROCFS 1 )
+set( EC_HAVE_DLFCN_H 1 )
+set( EC_HAVE_DLADDR 1 )
+set( EC_HAVE_AIOCB 1 )
+set( EC_HAVE_AIOCB64 1 )
+
+# Disable relative rpaths as aprun does not respect it
+set( ENABLE_RELATIVE_RPATHS OFF CACHE STRING "Disable relative rpaths" FORCE )
+
+####################################################################
+# COMPILER
+####################################################################
+
+set( ECBUILD_FIND_MPI ON )
+set( ECBUILD_TRUST_FLAGS ON )
+
+####################################################################
+# Compiler FLAGS
+####################################################################
+
+# General Flags (add to default)
+
+set(ECBUILD_Fortran_FLAGS "-g")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -qopenmp-threadprivate compat")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -assume byterecl")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -convert big_endian")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -traceback")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -align array64byte")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -warn nounused,nouncalled")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -march=core-avx2")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -finline-functions")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -finline-limit=1500")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -Winline")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -no-fma")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -assume realloc_lhs")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -fp-model precise")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -ftz")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -fp-speculation=safe")
+set(ECBUILD_Fortran_FLAGS "${ECBUILD_Fortran_FLAGS} -fast-transcendentals")
+
+####################################################################
+# Additional compiler flags for SYCL offload via CUDA backend
+####################################################################
+
+# Additional Intel DPCPP compiler for SYCL offload
+set(CMAKE_CXX_COMPILER "/home/nams/opt/dpcpp/bin/clang++")
+
+# Initial set of flags to things going with a custom DPCPP install on AC
+set(CMAKE_CXX_FLAGS "-O3 -L/home/nams/opt/dpcpp/lib -fopenmp -lstdc++")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsycl-early-optimizations -fsycl")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsycl-targets=nvptx64-nvidia-cuda")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Xsycl-target-backend --cuda-gpu-arch=sm_80")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -I/usr/local/apps/intel/2021.4.0/compiler/2021.4.0/linux/compiler/include")
```

Comment:
I think since we're now using `add_sycl_to_target`, the proper way would be to add the relevant ones to `SYCL_FLAGS` only. But because we're high-jacking the CMake module from a newer installation, I found it tricky to get to work reliably. So, just as an FYI and not necessary to change anything here.